### PR TITLE
Add panic placeholders for new float support added to Cryptol.

### PR DESCRIPTION
**Pull request by kquick**
This does not provide proper handling of the new float types, but adds panic calls on pattern matches to re-enable successful compilation and establish placeholders for future support.

### CODE REVIEW

**Review by yav — CHANGES_REQUESTED** [review]
The panics look fine, but I think we should change how we look up primitives.

File: src/Verifier/SAW/Cryptol.hs [hunk 1]
```diff
@@ -539,115 +543,136 @@ proveProp sc env prop =
...
-
-    _ -> panic "Unknown Cryptol primitive name" [C.unpackIdent nm]
-
-importPrimitive _ nm =
-  panic "Improper Cryptol primitive name" [show nm]
+importPrimitive sc n
+  | Just nm <- C.asPrim n, Just term <- Map.lookup nm (prelPrims <> floatPrims) = term sc
+  | Just nm <- C.asPrim n = panic "Unknown Cryptol primitive name" [show nm]
+  | otherwise = panic "Improper Cryptol primitive name" [show n]
+
```
Comment (yav) [inline src/Verifier/SAW/Cryptol.hs:550]: This is just stylistic, but I'd
  1. add separate top-level declaration for the map of translations (e.g. `allPrims = prelPrims <> floatPrims <> arrayPrim`)
  2. ~~I'd reduce the guards to 2 as that matches the logic more directly: either the name is a primitve or not.  In the case for the primitive I'd use a `case` to lookup in `allPrims` and `panic` if the primitive is missing.~~

File: src/Verifier/SAW/Cryptol.hs [hunk 2]
```diff
@@ -539,115 +543,136 @@ proveProp sc env prop =
...
+     --                            --              , first != next
+     --                            --              , lengthFromThenTo first next last == len) => [len]a
+
+   , ("error",        (scGlobalDef, "Cryptol.ecError"))       -- {at,len} (fin len) => [len][8] -> at -- Run-time error
+   , ("random",       (scGlobalDef, "Cryptol.ecRandom"))      -- {a} => [32] -> a -- Random values
+   , ("trace",        (scGlobalDef, "Cryptol.ecTrace"))       -- {n,a,b} [n][8] -> a -> b -> b
+   , ("Array::arrayConstant",(scGlobalDef, "Cryptol.ecArrayConstant")) -- {a,b} b -> Array a b
+   , ("Array::arrayLookup",  (scGlobalDef, "Cryptol.ecArrayLookup")) -- {a,b} Array a b -> a -> b
+   , ("Array::arrayUpdate",  (scGlobalDef, "Cryptol.ecArrayUpdate")) -- {a,b} Array a b -> a -> b -> Array a b
+   ])
```
Comment (yav) [inline src/Verifier/SAW/Cryptol.hs:658]: This is incorrect.  The `Array` primitives are in their own module, so we should make a separate map for them.  Their names are of the form `PrimIdent arrayName x`.  In general, there shouldn't be any `::` in the names of primitives.

By the way, I just noticed that we don't seem to have a convenience short-cut for the `Array` module (i.e., `arrayPrim = PrimIdent arrayName`), so feel free to add it to `Crytpol.Util.Ident`.  I am fine if you just add it without a pull request, but whatever works for you.

**Review by yav — CHANGES_REQUESTED** [review]
We should definitely fix the names of the `Array` primitives as that will break Andre's code.

On the stylistic issue, it's up to you.

**Comment by brianhuffman** [comment]
Superseded by #26.

**Comment by kquick** [comment]
Sorry @brianhuffman , looks like I had forgotten to push my latest patch set (I had run into the RecordSet changes in verifying those) and you had to redo that set of changes.  Thanks for resolving this in #26.

**Pull request closed without merging**